src/datasets/io/generator.py

```diff
@@ -24,7 +24,7 @@ def __init__(
             **kwargs,
         )
 
-    def read(self):
+    def read(self, streaming: bool = False):
```

Comment:
This is a new param, so let's move to the end.

Reply:
Thank you @mariosasko for your feedback.
I am not sure I understand what I need to do here. Can you please help me on this.
Also a little help on generating the test module would be appreciated.
Thank you!

Reply:
Oops, I didn't notice that this is the only param in the signature 🙂.